### DISCUSSION TITLE
in-cluster connection suddenly failed on one of our K8S clusters

### DISCUSSION THREAD

**Discussion by michazt** [argoproj/argo-cd#8760] [Q&A]
Hello all,

we have six K8S Clusters which are bootstrapped via a bash shell script (which uses kustomize, helm and some yaml files of our own) to get a similar setup on every cluster. One big part is ArgoCD on every cluster that handles the rest of our application setup in the clusters. 

On one of these clusters ArgoCD suddenly has no connection to the in-cluster resource and is not able to deploy any changes. We get the following error on this cluster:
argocd cluster list --server argocd.dev-example.com:44444 --insecure

```
SERVER                           NAME         VERSION     STATUS      MESSAGE
https://kubernetes.default.svc   in-cluster   1.21        Failed      failed to sync cluster https://10.96.0.1:443: failed to load initial state of resource Secret: The provided continue parameter is too old to display a consistent list result. You can start a new list without the continue parameter, or use the continue token in this response to retrieve the remainder of the results. Continuing with the provided token results in an inconsistent list - objects that were created, modified, or deleted between the time the first chunk was returned and now may show up in the list.
```

We are using
ArgoCD      v2.2.0+6da92a8
Kubernetes  v1.21.6
Go Version  go1.16.11
Helm        v3.7.1+g1d11fcb
kustomize   v4.2.0 2021-06-30T22:49:26Z

The other clusters are running fine and do not have any problems. On every cluster we have this in-cluster resource in ArgoCD that was somehow automatically created by our bootstrapping mechanism. All six clusters are running for about a year now without any problems with ArgoCD. On the problematic cluster we had some memory issues on the master control plane nodes and needed to restart all three of them. As far as I know this happened three times in the last two months. My guess is that something got out of order because of these restarts.
But apart from this problem the cluster with the problem runs fine and we can access it with kubectl and all the deployed application are working. But we can not apply any changes via ArgoCD anymore...

Hopefully someone here can give us some hints on how to further debug this issue? We tried to understand how this in-cluster resource is created but couldn't find it in any yaml files that are created via our bootstrapping mechanism with helm and kustomize. When I try to delete the in-cluster it gets recreated instantly by ArgoCD and is still in this failed state. 

One of my guesses is that a token or secret could not be changed accordingly? If so where and how could we find and reset this?

Or it is something completely different we do not think of?

Any help on this would be appreciated. And if you need further information let me know and I try to provide it. Unfortunately I can not provide the complete created yaml file because it contains some company confidential stuff I do not want to post in public. (And it is 4100 lines long, so it is not easy to find all confidential stuff to obfuscate them.) But I can provide it in parts if needed.

Regards,
Michael

**Comment by ericdurandsymphony** [comment]
I get this error when I create lots of Secrets in the cluster. A lot, like 60k.

**Reply by michazt** [reply]
@ericdurandsymphony , thanks for your answer but how do you repair/resolve this state then?

During some research we came up with a solution. We did set a higher value (default is 5 minutes) for --etcd-compaction-interval on the K8S apiserver configuration. With a higher value the problem vanished and ArgoCD was able to connect again to the in-cluster. But we still need to investigate
1. why is this happening only on one Cluster. The other Clusters have a higher load and have more running PODs and we do not need to change this values. So this is still a mystery to us.
2. what is the best value for --etcd-compaction-interval configuration in our situation
When I have further answers I post it here but this is at least resolved though I still have some questions.

**Reply by ericdurandsymphony** [reply]
I did not find a solution, I was doing some scale tests that were a little above our needs, so I backed out (removed some secrets) and it worked again.

**Reply by michazt** [reply]
Thx for the info. I think this kind of problems have nothing to do with ArgoCD in particular. It seems on certain circumstances you need to tweak K8S. In my case the higher value for etcd-compaction-interval solved it but still I have no insight what caused it.
